Retry failed codesign commands to work around Apple's flaky secure
timestamp service.

--- src/codesign.sh.orig
+++ src/codesign.sh
@@ -23,8 +23,10 @@ for f in "$@"; do
         chown "$SUDO_USER" "$f" "$DIR"
     fi
 
-    $SUDO /usr/bin/codesign --sign "$CODESIGN_ID" --identifier=org.macports.base --options=runtime --timestamp --verbose "$f"
-
+    while ! $SUDO /usr/bin/codesign --sign "$CODESIGN_ID" --identifier=org.macports.base --options=runtime --timestamp --verbose "$f"
+    do
+        sleep 1
+    done
     if [ -n "$SUDO" ]; then
         chown "$FILE_OWNER" "$f"
         chown "$DIR_OWNER" "$DIR"
